pkg/cli/validate.go

```diff
@@ -0,0 +1,191 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cli
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strings"
+	"text/tabwriter"
+	"time"
+
+	jvspb "github.com/abcxyz/jvs/apis/v0"
+	"github.com/abcxyz/jvs/client-lib/go/client"
+	"github.com/abcxyz/jvs/pkg/config"
+	"github.com/spf13/cobra"
+)
+
+// validateCmdOptions holds all the inputs and flags for the validate subcommand.
+type validateCmdOptions struct {
+	config *config.CLIConfig
+
+	token string
+}
+
+// newValidateCmd creates a new subcommand for validating tokens.
+func newValidateCmd(cfg *config.CLIConfig) *cobra.Command {
+	opts := &validateCmdOptions{
+		config: cfg,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "validate",
+		Short: "Validate the input token",
+		Long: strings.Trim(`
+Validate the given justification token passed via token flag or pipe.
+The output will be a tablized view of validation result, token justification
+and standard claims put together, or any errors that occured.
+
+For example:
+
+    # Validate the justification token string
+    jvsctl validate --token "example token string"
+
+    # Validate the justification token read from pipe
+    cat token.txt | jvsctl token --token -
+
+    # Output
+	-------RESULT-------
+	validated!
+	breakglass  false
+	----JUSTIFICATION----
+	explanation  "test"
+	---STANDARD CLAIMS---
+	aud  ["dev.abcxyz.jvs"]
+	iat  "2022-01-01T00:00:00Z"
+	iss  "jvsctl"
+	jti  "test-jwt"
+	nbf  "2022-01-01T00:00:00Z"
+	sub  "jvsctl"
+`, "\n"),
+		Args: cobra.ExactArgs(0),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runValidateCmd(cmd, opts, args)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVarP(&opts.token, "token", "t", "",
+		"The JVS token that needs validation")
+	cmd.MarkFlagRequired("token") //nolint // not expect err
```

Comment:
Can we link to the issue (or create one) to output in raw json? E.g. `--format=json`.